[Muhammad Faruq] iP

muhammad-faruq opened the pull request:
### BoB
> "I'm ready, I'm ready - SpongeBob SquarePants ([source](https://en.wikipedia.org/wiki/SpongeBob_SquarePants_(character)))

BoB frees your mind of having to remember things you need to do. It's,

- text-base
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. Download it [here](https://github.com/muhammad-faruq/ip)
2. Double-click it
3. Add your tasks using the specified commands
4. PROFIT!! 🤩 

Best part is... **ITS FREE!!**

Features:
- [x] Managing basic tasks
- [x] Searching through current tasks
- [x] Deletion of task
- [ ] Managing Deadlines (coming soon)
- [ ] Reminders (coming soon)

If you practicing Java, you can use this to practice!! Here's the `main` method:
```java
// BoB was derived from Duke
public class Duke{
    public static void main(String[] args) {
        Duke duke = new Duke("LOCATION_TO_STORE_LIST");
        duke.run();
    }
}
```

Commit:
Add Gradle support

muhammad-faruq committed:
Duke changed to BoB at level 1

muhammad-faruq committed:
Level-2 but i forgot to push before doing level 3

muhammad-faruq committed:
Level 3 submission

muhammad-faruq committed:
test

muhammad-faruq committed:
resubmission of level 1

muhammad-faruq committed:
resubmission of level-2

muhammad-faruq committed:
resubmission of level-3

muhammad-faruq committed:
level-4 submission

muhammad-faruq committed:
A-textUiTestingA-textUiTesting

muhammad-faruq committed:
stupid amounts of refactoring for Level-5

muhammad-faruq committed:
level 6

muhammad-faruq committed:
A-Enums

muhammad-faruq committed:
Add save functionality

muhammad-faruq committed:
Add Date-Time formatting

muhammad-faruq committed:
Add Date-Time functionality (Level-8)

muhammad-faruq committed:
Refactor to use more OOP

muhammad-faruq committed:
Refactor to use more OOP

muhammad-faruq committed:
Refactor into packages

muhammad-faruq committed:
Add JUnit Tests

muhammad-faruq committed:
Bug fix for no data.txt

muhammad-faruq committed:
Add JavaDoc for all class and methods

muhammad-faruq committed:
Merge branch 'branch-A-JavaDoc'

muhammad-faruq committed:
Refactor to comply with coding standards

muhammad-faruq committed:
Merge conflicts with refactoring code

muhammad-faruq committed:
Add Find functionality

muhammad-faruq committed:
Merge conflicts for Find functionality

muhammad-faruq committed:
Merge branch 'add-gradle-support'

muhammad-faruq committed:
Add Gradle support

muhammad-faruq committed:
Fix Checkstyle

muhammad-faruq committed:
Add GUI to Duke

muhammad-faruq committed:
Add varargs

muhammad-faruq committed:
Merge branch 'branch-A-Varargs'

muhammad-faruq committed:
Fix javafx bugs. Allow for responsive text overflow.

tanjiaxian99 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+
+    private Duke duke = new Duke(".\\src\\main\\resources\\data\\data.txt");
+
+    @Override
```
Perhaps a comment here for the Start() method?

tanjiaxian99 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    public String getResponse(String input) {
```
Perhaps a JavaDoc comment here to describe what getResponse does?

muhammad-faruq replied:
Yea i realize the bulk of this is from the GUI update where i have not written javadocs for

tanjiaxian99 commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Test
+    void testToString() {
```
Maybe consider more consistent naming between test methods? Such as removing "test" from the names of tests.

tanjiaxian99 commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,42 @@
+    private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/surge.jpg"));
+
+    @FXML
+    public void initialize() {
```
Maybe a JavaDoc comment here to describe what the method does?

tanjiaxian99 reviewed (commented):
Very few coding violations. Well done! LGTM.

djiangel commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(new ArrayList< Task >());
```
I think there are extra spaces for ArrayList< Task >().

djiangel commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+     * @return The additional information of the task which is the deadline of the task.
+     */
+    @Override
+    public String getAdditionalInfo() {
```
Maybe can just name it something more specific?

djiangel reviewed (commented):
LGTM! Other than those two small points I commented on (:

muhammad-faruq committed:
Bug fixes for saving and add javadocs for classes.

Some issues with the saving of data when building fatJar using gradle.
Changed filepath to fix the bug.

Javadocs was added for the remainder of the methods and checkstyle was
cleaned up for all java files.

muhammad-faruq committed:
Add assert checks

Assert checks is added in Duk Exception, Parser, Storage and TaskList.

muhammad-faruq committed:
Improve code quality

Changed some magic strings/ numbers and refactored some other code to
improve readability

muhammad-faruq committed:
Merge pull request #1 from muhammad-faruq/branch-A-CodeQuality

Improve code quality

muhammad-faruq committed:
Merge branch 'master' into branch-A-Assertions

muhammad-faruq committed:
Merge pull request #2 from muhammad-faruq/branch-A-Assertions

Add assert checks

muhammad-faruq committed:
Add test github actions

muhammad-faruq committed:
Create github-actions.yml

muhammad-faruq committed:
Merge branch 'master' of https://github.com/muhammad-faruq/ip

muhammad-faruq committed:
Add github actions

muhammad-faruq committed:
Add Do After Task

Do after is a task that takes in a task that needs to be done after
another task.
For example: doafter eat dinner /after getting home

muhammad-faruq committed:
Improve GUI for Duke

Added background image, and wrapping dialog text box.

muhammad-faruq committed:
Ad Javadoc

muhammad-faruq committed:
Add UI image

Added product screenshot and additional supporting documents

muhammad-faruq committed:
Add javadocs for javafx components

muhammad-faruq committed:
Delete ui

Name was wrong

muhammad-faruq committed:
Add Ui.png

muhammad-faruq committed:
Add user guide

muhammad-faruq committed:
Fix version bugs

muhammad-faruq committed:
Improve README

muhammad-faruq committed:
readme improvements
